backends/gstreamer/render-unix/lib.rs

```diff
@@ -0,0 +1,193 @@
+        match gl_context {
+            GlContext::Egl(context) => {
+                let display = unsafe { gst_gl::GLDisplayEGL::new_with_egl_display(display_native) };
+                if let Some(display) = display {
+                    let context = unsafe {
+                        gst_gl::GLContext::new_wrapped(
+                            &display,
+                            context,
+                            gst_gl::GLPlatform::EGL,
+                            gst_gl::GLAPI::ANY,
+                        )
+                    };
+
+                    if let Some(context) = context {
+                        if !(context.activate(true).is_ok() && context.fill_info().is_ok()) {
+                            let cat = gst::DebugCategory::get("default").unwrap();
+                            gst_warning!(cat, "Couldn't fill the wrapped app GL context")
+                        }
+                        return Some(RenderUnix {
+                            display: display.upcast(),
+                            app_context: context,
+                            gst_context: Arc::new(Mutex::new(None)),
+                            gl_upload: Arc::new(Mutex::new(None)),
+                        });
+                    }
+                }
+
+                None
+            }
+            _ => None,
+        }
+    }
+}
+
+impl Render for RenderUnix {
+    fn is_gl(&self) -> bool {
+        true
+    }
+
+    fn build_frame(&self, buffer: gst::Buffer, info: gst_video::VideoInfo) -> Result<Frame, ()> {
+        if self.gst_context.lock().unwrap().is_none() && self.gl_upload.lock().unwrap().is_some() {
+            *self.gst_context.lock().unwrap() =
+                if let Some(glupload) = self.gl_upload.lock().unwrap().as_ref() {
+                    glupload
+                        .get_property("context")
+                        .or_else(|_| Err(()))?
+                        .get::<gst_gl::GLContext>()
+                } else {
+                    None
+                };
+        }
+
+        //let sync_meta = buffer.as_ref().get_meta::<gst_gl::GLSyncMeta>().unwrap();
+        //sync_meta.set_sync_point(self.gst_context.borrow().as_ref().unwrap());
+
+        let frame =
+            gst_video::VideoFrame::from_buffer_readable_gl(buffer, &info).or_else(|_| Err(()))?;
+
+        Frame::new(
+            info.width() as i32,
+            info.height() as i32,
+            Arc::new(GStreamerBuffer { frame }),
+        )
+    }
+
+    fn build_video_sink(
+        &self,
+        appsink: &gst::Element,
+        pipeline: &gst::Element,
+    ) -> Result<(), PlayerError> {
+        if self.gl_upload.lock().unwrap().is_some() {
+            return Err(PlayerError::Backend(
+                "render unix already setup the video sink".to_owned(),
+            ));
+        }
+
+        let vsinkbin = gst::ElementFactory::make("glsinkbin", "servo-media-vsink")
```

Comment:
Would it make sense to add a check for this element in `set_gl_params` like the one that we have in [GStreamerPlayer.setup](https://github.com/servo/media/blob/master/backends/gstreamer/player.rs#L492)?